pros/src/adi/analog.rs

```diff
@@ -0,0 +1,116 @@
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+use crate::error::bail_on;
+
+use pros_sys::PROS_ERR;
+
+pub struct AdiAnalogIn {
+    port: u8,
+}
+
+impl AdiAnalogIn {
+    /// Create an AdiAnalogIn without checking if it is valid.
+    /// 
+    /// # Safety
+    /// 
+    /// The port must be above 0 and below [`pros_sys::NUM_ADI_PORTS`].
+    pub unsafe fn new_unchecked(slot: AdiSlot) -> Self {
+        Self {
+            port: slot as u8
+        }
+    }
+    
+    /// Create an AdiAnalogIn, throwing an error if the port is invalid.
```

Comment:
This should either say "panicking if the port is invalid" or do something like this:
```
/// Creates an AdiAnalogIn.
/// # Panics
/// Panics if the given port is invalid
```

Typically throws an error is not used to refer to panics in rust code.